src/vm/i386/asmhelpers.S

```diff
@@ -821,10 +821,10 @@ NESTED_END DelayLoad_MethodCall, _TEXT
 //  to optionally patch the target of the jump so that we do not take this slow path again.
 //
 NESTED_ENTRY VirtualMethodFixupStub, _TEXT, NoHandler
-    // Pop the return address. It points right after the call instruction in the thunk.
-    pop     eax
+    // Get the return address. It points right after the call instruction in the thunk.
+    mov     ebx, esp
```

Comment:
This is wrong. It should get return address, not return address adress on the stack. So the next calculation of the address of the thunk returns some address on the stack instead of the address of the code.